recipe/thirdparty/tpmodels/apiInterface.go

```diff
@@ -24,22 +24,43 @@ import (
 )
 
 type APIInterface struct {
-	AuthorisationUrlGET      *func(provider TypeProvider, options APIOptions, userContext supertokens.UserContext) (AuthorisationUrlGETResponse, error)
-	SignInUpPOST             *func(provider TypeProvider, code string, authCodeResponse interface{}, redirectURI string, options APIOptions, userContext supertokens.UserContext) (SignInUpPOSTResponse, error)
-	AppleRedirectHandlerPOST *func(code string, state string, options APIOptions, userContext supertokens.UserContext) error
+	AuthorisationUrlGET *func(provider TypeProvider, clientType *string, tenantId *string, redirectURIOnProviderDashboard string, options APIOptions, userContext supertokens.UserContext) (AuthorisationUrlGETResponse, error)
+	SignInUpPOST        *func(provider TypeProvider, clientType *string, tenantId *string, input TypeSignInUpInput, options APIOptions, userContext supertokens.UserContext) (SignInUpPOSTResponse, error)
+
+	AppleRedirectHandlerPOST *func(formPostInfoFromProvider map[string]interface{}, options APIOptions, userContext supertokens.UserContext) error
+	ProvidersForTenantGET    *func(tenantId string, userContext supertokens.UserContext) (ProvidersForTenantGetResponse, error)
+}
+
+type ProvidersForTenantGetResponse struct {
+	OK *struct {
+		Providers []struct {
+			ID              string `json:"id"`
+			Name            string `json:"name"`
```

Comment:
this should be optionally sent to the frontend because if we send the statically defined list as well, then those don't have names.

Reply:
Done